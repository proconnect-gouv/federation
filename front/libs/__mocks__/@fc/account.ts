import React from 'react';

export const AccountProvider = jest.fn();

export const AccountContext = React.createContext(expect.any(Object));
