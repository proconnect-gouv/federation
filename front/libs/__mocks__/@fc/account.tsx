import React from 'react';

export const AccountProvider = jest.fn(({ children }) => (
  <div>
    <div>AccountProvider</div>
    <div>{children}</div>
  </div>
));

export const AccountContext = React.createContext(expect.any(Object));
