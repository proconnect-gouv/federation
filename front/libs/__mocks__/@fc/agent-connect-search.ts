import * as React from 'react';

export const AgentConnectSearchContext = React.createContext({
  hasSearched: false,
  payload: {
    csrfToken: 'csrf-token-mock',
    identityProviders: [],
    isLoaded: false,
    ministries: [],
    serviceProviderName: undefined,
  },
  searchResults: [],
  setSearchTerm: () => {},
});

export const useSearchResults = jest.fn(() => ({
  searchResults: [
    {
      identityProviders: [
        {
          active: true,
          display: true,
          name: 'mock-idp-name-1.1',
          uid: 'mock-idp-uid-1.1',
        },
        {
          active: true,
          display: true,
          name: 'mock-idp-name-1.2',
          uid: 'mock-idp-uid-1.2',
        },
      ],
      ministry: {
        id: 'mock-ministry-id-1',
        identityProviders: ['mock-idp-uid-1'],
        name: 'mock-ministry-name-1',
        sort: 1,
      },
    },
    {
      identityProviders: [
        {
          active: true,
          display: true,
          name: 'mock-idp-name-2.1',
          uid: 'mock-idp-uid-2.1',
        },
        {
          active: true,
          display: true,
          name: 'mock-idp-name-2.2',
          uid: 'mock-idp-uid-2.2',
        },
      ],
      ministry: {
        id: 'mock-ministry-id-2',
        identityProviders: ['mock-idp-uid-2'],
        name: 'mock-ministry-name-2',
        sort: 0,
      },
    },
  ],
  showNoResults: false,
  showResults: true,
}));

export const useOnSearch = jest.fn(() => jest.fn());
