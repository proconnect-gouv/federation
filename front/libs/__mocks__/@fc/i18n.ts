export const I18nService = {
  initialize: jest.fn(),
};

export const t = jest.fn((key) => key);
