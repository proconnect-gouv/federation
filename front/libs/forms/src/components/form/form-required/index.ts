/* istanbul ignore file */

// declarative file
export * from './form-required.component';
