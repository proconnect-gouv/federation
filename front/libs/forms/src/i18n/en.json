{
  "Form.button.copy": "Copier",
  "Form.submit": "Submit",
  "Form.reset": "Reset the form",
  "Form.message.valid": "Validation text",
  "Form.message.required": "This field is required",
  "Form.message.requestFailed": "The message could not be sent",
  "Form.message.requiredFields": "All fields with an asterisk are required.",
  "Form.select.placeholder": "Select an option",
  "Form.textarea.maxlength": " / {maxLength} characters",
  "Form.multiple.add": "Add a new value",
  "Form.multiple.remove": "Remove this value"
}
