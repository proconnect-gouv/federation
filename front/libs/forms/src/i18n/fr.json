{
  "Form.button.copy": "Copier",
  "Form.submit": "Envoyer ces informations",
  "Form.reset": "Effacer le formulaire",
  "Form.message.valid": "Texte de validation",
  "Form.message.required": "Ce champs est obligatoire",
  "Form.message.requestFailed": "Le message n’a pas pu être envoyé",
  "Form.message.requiredFields": "Tous les champs avec un astérisque sont obligatoires.",
  "Form.select.placeholder": "Sélectionner une option",
  "Form.textarea.maxlength": " / {maxLength} caractères",
  "Form.FORM_ERROR": "Une erreur s'est produite lors de la soumission du formulaire.",
  "Form.multiple.add": "Ajouter une valeur",
  "Form.multiple.remove": "Supprimer cette valeur"
}
