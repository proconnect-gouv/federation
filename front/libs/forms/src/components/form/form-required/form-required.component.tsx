import React from 'react';

import { t } from '@fc/i18n';

export const FormRequiredMessageComponent = React.memo(() => (
  <p className="fr-text--sm fc-text-align--left-md-center">{t('Form.message.requiredFields')}</p>
));

FormRequiredMessageComponent.displayName = 'FormRequiredMessageComponent';
