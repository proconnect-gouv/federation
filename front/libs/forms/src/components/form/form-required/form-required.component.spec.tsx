import { render } from '@testing-library/react';

import { t } from '@fc/i18n';

import { FormRequiredMessageComponent } from './form-required.component';

describe('FormRequiredMessageComponent', () => {
  it('should match the snapshot', () => {
    // When
    const { container } = render(<FormRequiredMessageComponent />);

    // Then
    expect(container).toMatchSnapshot();
    expect(t).toHaveBeenCalledWith('Form.message.requiredFields');
  });
});
