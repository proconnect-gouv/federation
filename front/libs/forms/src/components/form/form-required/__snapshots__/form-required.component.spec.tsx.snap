// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormRequiredMessageComponent should match the snapshot 1`] = `
<div>
  <p
    class="fr-text--sm fc-text-align--left-md-center"
  >
    Form.message.requiredFields
  </p>
</div>
`;
